xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3647,6 +3645,14 @@ void CVideoPlayer::SetSpeed(float speed)
   
   m_newPlaySpeed = speed * DVD_PLAYSPEED_NORMAL;
   SetPlaySpeed(speed * DVD_PLAYSPEED_NORMAL);
+  if (m_newPlaySpeed == DVD_PLAYSPEED_NORMAL && m_playSpeed != DVD_PLAYSPEED_NORMAL)
```

Comment:
Why does it need to be before SetPlaySpeed? I thought that was what you ment, when you said the code is async and it doesn't matter in which order it get's called? Just trying to understand things here.

I like what you did to the code format, will test and push the changes.